File: stacks/evaluate_reverse_polish_notation/eval_polish_notation.go
// Package rpn provides functions to evaluate expressions written in Reverse Polish Notation.
package rpn

import "strconv"

func EvalRPN(tokens []string) int {
	stack := []int{}
	for _, token := range tokens {
		switch token {
		case "+":
			a := stack[len(stack)-1]
			b := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, b+a)
		case "-":
			a := stack[len(stack)-1]
			b := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, b-a)
		case "*":
			a := stack[len(stack)-1]
			b := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, b*a)
		case "/":
			a := stack[len(stack)-1]
			b := stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, b/a)
		default:
			i, _ := strconv.Atoi(token)
			stack = append(stack, i)
		}
	}
	return stack[0]
}

File: stacks/evaluate_reverse_polish_notation/eval_polish_notation.py
def eval_rpn(tokens):
    stack = []
    for token in tokens:
        if token == "+":
            stack.append(stack.pop() + stack.pop())
        elif token == "-":
            a = stack.pop()
            b = stack.pop()
            stack.append(b - a)
        elif token == "*":
            a = stack.pop()
            b = stack.pop()
            stack.append(a * b)
        elif token == "/":
            a = stack.pop()
            b = stack.pop()
            stack.append(int(b / a))
        else:
            stack.append(int(token))
    return stack[0]
